docs/tutorial/quick-start.md

```diff
@@ -0,0 +1,92 @@
+
+***This must be done for GitHub runner charm to function correctly.***
+
+The GitHub runner charm uses GitHub APIs related to self-hosted runners. Some of the APIs will only be functional after a self-hosted runner registration token is requested for the repository for the first time. If a registration token is requested for the repository before then this can be skipped.
+
+The registration token can be requested by calling the [GitHub API to request a self-hosted runner registration token](https://docs.github.com/en/rest/actions/self-hosted-runners?apiVersion=2022-11-28#create-a-registration-token-for-a-repository). Alternatively, a registration token can be requested by visiting the "New self-hosted runner" page for the repository (`https://github.com/{OWNER}/{REPO}/settings/actions/runners/new`). This can be done by following the instruction to the 4th step provided [here](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/adding-self-hosted-runners#adding-a-self-hosted-runner-to-a-repository).
+
+### Deploy the GitHub runner charm
+
+The charm requires a GitHub personal access token with `repo` access, which can be created with the instructions [here](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic).
+
+Once the personal access token is created, the charm can be deployed with:
+
+```shell
+juju deploy github-runner --constraints="cores=4 mem=16G" --config token=<TOKEN> --config path=<OWNER/REPO>
+```
+
+Replacing the `<TOKEN>` with the personal access token, and `<OWNER/REPO>` the GitHub account name and GitHub repository separated with `/`.
+
+The `--constraints` option for the `juju deploy` set the resource used for the juju machine hosting the charm application. This is used to accommodate different size of self-hosted runner.
+
+Once the charm is active status, visit the runner page for the GitHub repository (`https://github.com/{OWNER}/{REPO}/settings/actions/runners`) according to the instructions [here](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/using-self-hosted-runners-in-a-workflow#viewing-available-runners-for-a-repository). A single new runner should be available as it is the default number of self-hosted runners created.
+
+The charm would regularly goes into maintain status to managed the self-hosted runners. For example, replacing a self-hosted runner that was used up by a GitHub Actions job.
+
+### Ensure GitHub repository settings are secure
+
+For public repository, [arbitrary code execution with the self-hosted runners are possible](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners#self-hosted-runner-security). To combat this, the charm enforce a set of setting for the repository to ensure the code executed in the self-hosted runner are reviewed by someone trusted.
```

Comment:
s/repository/repositories
s/with the self-hosted runners are possible/within the self-hosted runners is possible
s/enforce/enforces
s/for the repository/for the repositories
s/the code executed in the self-hosted runner are reviewed by someone trusted/the code executed is reviewed by someone trusted